src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -29,48 +27,267 @@ export const CollectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            return activeLotData.saleArtwork.bidder_positions_count
+          }
+
+          return null
+        },
       },
       lotWatcherCount: {
         type: GraphQLInt,
         description: "Lot watcher count on lots open for bidding",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            return artwork.recent_saves_count
+          }
+
+          return null
+        },
       },
       registrationEndsAt: {
         type: GraphQLString,
         description: "Pending auction registration end time",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            const registrationEndAtDate =
+              activeLotData.sale.registration_ends_at &&
+              new Date(activeLotData.sale.registration_ends_at)
+
+            if (registrationEndAtDate && registrationEndAtDate > NOW) {
+              return registrationEndAtDate.toISOString()
+            }
+          }
+
+          return null
+        },
       },
       lotClosesAt: {
         type: GraphQLString,
         description: "Pending auction lot end time for bidding",
+        resolve: async (artwork, {}, ctx) => {
+          const isInSale = artwork.sale_ids?.length > 0
+
+          if (
+            !checkFeatureFlag("emerald_signals-auction-improvements", ctx) ||
+            !isInSale
+          ) {
+            return null
+          }
+
+          const activeLotData = await getActiveSaleArtwork(
+            {
+              artworkId: artwork.id,
+              saleIds: artwork.sale_ids,
+            },
+            ctx
+          )
+
+          if (activeLotData) {
+            const extendedBiddingEndAtDate =
```

Comment:
Hand-wave-y comment -> feels like this logic should actually live somewhere in Gravity w/ the bidding engine - single source of truth for this timestamp.

Also, I'm unclear why the `> NOW` is required - it seems like the correct timestamp should always be _either_ `extended_bidding_end_at` if available, or `end_at` otherwise. (Also - I think for OO sales that _don't_ employ cascading/popcorn - not sure if we do those but the code allows for it - each lot wouldn't even have an `end_at`, it'd just be `sale.end_at`). So I'd expect something like `extended_bidding_end_at || saleArtwork.end_at || sale.end_at` (hand-wave-y).

Additionally - we should be careful about checking timestamps across systems. These times are set within Gravity, but then `NOW` is an MP-server time - what if MP clock is off? I think there's likely a solution that doesn't require `NOW` which (in a very hand-wave-y/intuitive way) doesn't seem right to me.